pkg/cvo/sync_worker.go

```diff
@@ -668,8 +668,10 @@ func (w *SyncWorker) apply(ctx context.Context, payloadUpdate *payload.Update, w
 		return nil
 	})
 	if len(errs) > 0 {
-		err := cr.Errors(errs)
-		return err
+		if err := cr.Errors(errs); err != nil {
+			return err
+		}
+		return errs[0]
```

Comment:
We are only sending the first context error to syncOnce as you mentioned in the commit message. However in the code it is not clear to me that sending all the errors will cause any damage. 
```
		return w.syncOnce(ctx, work, maxWorkers, reporter)
			}()
			if err != nil {
				// backoff wait
				// TODO: replace with wait.Backoff when 1.13 client-go is available
				consecutiveErrors++
```

Reply:
> However in the code it is not clear to me that sending all the errors will cause any damage.

We could probably return all of them if we wanted, but `errs` is an `[]error` and `apply()` returns an `error`, so we'd need to do something to collapse this down to a single `error`.  I'm fine trying with `errors.NewAggregate(errs)` instead if you prefer, and don't see a lot of space between the two approaches.  We will already be logging the unfiltered list under `All errors were cancellation errors...`, so I don't see a problem with excluding some of the additional context errors in other logging fed by the `syncOnce` return value.

Reply:
Thanks for the explanation. I missed that part when I was reviewing the PR. `func NewAggregate()` sounds better because we are not just returning the first error.

Reply:
So the goal is to return context errors if they exist and not ignore them. So either we just use `NewAggregate` and not use `cr.Errors` as we do not want to filter context errors or create a function for this specific purpose. Is there benefit of current code VS just using `NewAggregate`?

Reply:
> Is there benefit of current code VS just using NewAggregate?

The context errors are from the individual graph workers, so depending on how many graph workers are aborted this way you can get a number of stuttering values.  For example, looking at one of the logs attached to the bug:

```console
$ grep 'Result of work.*update was cancelled' | head -n2
Logs_for_both_environments/unsuccessful_oc_logs_cluster_version_operator.log | head
I0524 16:08:36.953283       1 task_graph.go:611] Result of work: [update was cancelled at 0 of 573]
I0524 16:17:14.532134       1 task_graph.go:611] Result of work: [Cluster operator machine-config is still updating update was cancelled at 196 of 573 Cluster operator openshift-apiserver is reporting a failure: APIServerDeploymentDegraded: 1 of 3 requested instances are unavailable update was cancelled at 196 of 573]
```

Formatting there is not super clear, but the 16:17 entry had four workers fail:

* Cluster operator machine-config is still updating
* update was cancelled at 196 of 573
* Cluster operator openshift-apiserver is reporting a failure: APIServerDeploymentDegraded: 1 of 3 requested instances are unavailable
* update was cancelled at 196 of 573

I don't see a need to include a bunch of stutters in cases where the only worker issues were context failures.  But if you feel a strong preference for `NewAggregate` I can use that instead ;).

Reply:
You are right as per https://github.com/openshift/cluster-version-operator/blob/master/pkg/payload/task_graph.go#L592-L599 I was expecting below
```
I0524 16:17:14.532134       1 task_graph.go:611] Result of work: [Cluster operator machine-config is still updating update was cancelled at 196 of 573 Cluster operator openshift-apiserver is reporting a failure: APIServerDeploymentDegraded: 1 of 3 requested instances are unavailable update was cancelled at 196 of 573
```
The only question remaining is whether we will miss out any useful information if we do not report all the errors but `return errs[0]` ?

Reply:
We won't, because the `All errors were cancellation errors...` log line will contain the full slice.